python/lsst/drp/tasks/wcsfit.py

```diff
@@ -0,0 +1,1019 @@
+                                 "XPoly": {"OrderX": self.config.devicePolyOrder,
+                                           "SumOrder": True},
+                                 "YPoly": {"OrderX": self.config.devicePolyOrder,
+                                           "SumOrder": True},
+                                 "XMin": xMin, "XMax": xMax, "YMin": yMin, "YMax": yMax}
+            elif "identity" in component.lower():
+                componentDict = {"Type": "Identity"}
+            else:
+                raise ValueError(f"deviceModel component {component} is not supported")
+            inputDict[component] = componentDict
+
+        exposureModel = {"Type": "Composite", "Elements": self.config.exposureModel.list()}
+        inputDict["EXPOSURE"] = exposureModel
+        for component in self.config.exposureModel:
+            if "poly" in component.lower():
+                componentDict = {"Type": "Poly",
+                                 "XPoly": {"OrderX": self.config.exposurePolyOrder,
+                                           "SumOrder": "true"},
+                                 "YPoly": {"OrderX": self.config.exposurePolyOrder,
+                                           "SumOrder": "true"}}
+            elif "identity" in component.lower():
+                componentDict = {"Type": "Identity"}
+            else:
+                raise ValueError(f"exposureModel component {component} is not supported")
+            inputDict[component] = componentDict
+
+        inputYAML.addInput(yaml.dump(inputDict))
+        inputYAML.addInput("Identity:\n  Type:  Identity\n")
+
+        return inputYAML
+
+    def _addObjects(self, wcsf, inputCatalogRefs):
+        """Add science sources to the wcsfit.WCSFit object.
+
+        Parameters
+        ----------
+        wcsf : `wcsfit.WCSFit`
+            WCS-fitting object
+        inputCatalogRefs : `list`
+            List of DeferredDatasetHandles pointing to visit-level source
+            tables
+        """
+        for inputCatalogRef in inputCatalogRefs:
+            visit = inputCatalogRef.dataId['visit']
+
+            inputCatalog = inputCatalogRef.get(parameters={'columns': self.columns})
+            detectors = set(inputCatalog['detector'])
+
+            for detector in detectors:
+                detectorSources = inputCatalog[inputCatalog['detector'] == detector]
+
+                extensionIndex = self.extensionInfo[(self.extensionInfo['visit'] == visit)
+                                                    & (self.extensionInfo['detector'] == detector)].index[0]
+                sourceCat = detectorSources[self.sourceIndices[extensionIndex]]
+
+                d = {"x": sourceCat['x'].to_numpy(),
+                     "y": sourceCat['y'].to_numpy(),
+                     "xCov": sourceCat['xErr'].to_numpy()**2,
+                     "yCov": sourceCat['yErr'].to_numpy()**2,
+                     "xyCov": np.zeros(len(sourceCat))}
```

Comment:
Jointcal computed the xy err from the x and y errors and the shape moments (see `CcdImage:loadCatalog`): would a similar assumption be better here, than just setting them to zero?

Reply:
I tried implementing this, following the `loadCatalog` method, but I ran into some problems. There are a fair number (~1% maybe) of detections that have inferred covariances that are too high compared to the variance. I see this throws a warning in `jointcal`, and if I leave these objects in, the final chi2 is catastrophically worse. I could remove these from the fit; there may be a flag that helps. I don't know if the bad shape measurement is a symptom of other problems. @parejkoj, did you ever look into this?

Reply:
This is related to DM-7101 and DM-7102. I suspect that everything I know about it is available on those tickets (and possible ones linked from them).